src/compiler/parser.ts

```diff
@@ -1333,58 +1334,80 @@ module ts {
         }
 
         function parseComputedPropertyName(): ComputedPropertyName {
-            // PropertyName[Yield,GeneratorParameter] :
+            // PropertyName[Yield,GeneratorParameter,Await,AsyncParameter] : (Modified)
             //     LiteralPropertyName
             //     [+GeneratorParameter] ComputedPropertyName
-            //     [~GeneratorParameter] ComputedPropertyName[?Yield]
+            //     [+AsyncParameter] ComputedPropertyName
+            //     [~GeneratorParameter,~AsyncParameter] ComputedPropertyName[?Yield,?Await]
             // 
-            // ComputedPropertyName[Yield] :
-            //     [ AssignmentExpression[In, ?Yield] ]
+            // ComputedPropertyName[Yield,Await] :
+            //     [ AssignmentExpression[In,?Yield,?Await] ]
             //
             var node = <ComputedPropertyName>createNode(SyntaxKind.ComputedPropertyName);
             parseExpected(SyntaxKind.OpenBracketToken);
 
+            var explicitFlags = ParserContextFlags.None;
+            var ambientFlags = ParserContextFlags.None;
+            if (!inGeneratorParameterContext() && !inAsyncParameterContext()) {

```

Comment:
Per my grammar above, `[GeneratorParameter]` and `[AsyncParameter]` are mutually-exclusive. The ES parametric grammar does not currently have an idiom for expressing that more than a single parameter are unset, so I opted to comma delimit them for now. In any case, the expectation is that you are in either `[GeneratorParameter]`, `[AsyncParameter]`, or neither, but never both. This may change if we ever look at *async generators* but I think it may be quite a while before we consider those.